api/v1alpha1/openapi.yaml

```diff
@@ -2580,6 +2580,64 @@ components:
               - filePath
           required:
           - httpRef
+    ApplicationSpec:
+      type: object
+      description: "Defines the specification for an application"
+      properties:
+        name:
+          type: string
+          description: "The name of the application"
+        version:
+          type: string
+          description: "The version of the application"
+        description:
+          type: string
+          description: "A brief description of the application"
+        configRef:
+          type: string
+          description: "A reference to the name of a config which is used as the configuration for the application"
+        podmanCompose:
+          $ref: '#/components/schemas/PodmanComposeProviderSpec'
+      required:
+        - name
+        - configRef
+      oneOf: # extend this list with more providers to ensure that only one provider is used
+        - required: ["podmanCompose"]
+        # - required: ["quadlets"]
+    ConfigMounts:
+      type: array
+      items:
+        $ref: '#/components/schemas/ApplicationConfigRef'
+      description: |
+        An array of configuration mounts. 
+        Each item maps a configuration name reference to a mount point inside the application.
```

Comment:
A configuration name? what's that exactly where does it come from?

Reply:
I will flesh out the docs a bit but

https://github.com/flightctl/flightctl/blob/3e1633c7dc7a647e94f8847c08ee6ed4ae10c238/api/v1alpha1/openapi.yaml#L2493-L2499

```
      config:
        - name: base
          configType: GitConfigProviderSpec
          gitRef:
            repository: flightctl-demos
            targetRevision: main
            path: /demos/basic-nginx-demo/configuration
  ```

Reply:
I gues we could use `{ device.config[n].name }`

Reply:
I guess technically its a device.respository.name in the db.

Reply:
So I guess the question is this a  `ConfigMount` or `RepoMount`

Reply:
I will align with config
```
The name of the repository resource to use as the sync source
```

Reply:
But when it reaches the agent its a config right? :)

Reply:
The agent doesn't really know, it's just a set of files. Which have been mapped through ignition. But the server could also resolve those volume mount paths to device paths